app/models/services/locators/pubmed_paper_locator.rb

```diff
@@ -6,35 +6,31 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
     pubmed = Pubmed.new
     result = pubmed.get_uid_metadata(self.locator_id)
     if (data = result['result'][self.locator_id])
-      names = data['authors'].map {|a| a['name']}
-
-      existing_authors = Author.where(name: names)
-      existing_names = existing_authors.map(&:name)
-      new_authors = (names - existing_names).map {|a| Author.create name: a}
+      paper = self.import_data_to_paper(paper,data,'pubmed')
+    end
 
-      doi = data['elocationid']
-      if doi.blank?
-        doi = data['articleids'].find {|id| id['idtype'] == 'doi' }['value']
+    if paper.doi
+      crossref = Crossref.new
+      if result = crossref.get_doi_metadata(self.locator_id)
+        data = result['message']
+        paper = self.import_data_to_paper(paper,data,'crossref')
       end
+    end
 
-      doi = doi.sub(/^doi: /, "")
+    existing_paper = Paper.where( "title = ? or doi = ? and doi is not null", paper.title, paper.doi )
 
-      paper = Paper.create(
-        pubmed_id: data['uid'],
-        title: data['title'],
-        published_at: data['pubdate'],
-        authors: (existing_authors + new_authors),
-        abstract: pubmed.get_abstract(data['uid']),
-        doi: doi,
-        publication: data['source']
-      )
+    if existing_paper.present?
+      return existing_paper
+    elsif paper.save
       return paper
     else
-      paper = Paper.new
-      paper.errors.add(:locator_id, "is invalid; no paper found for searched Pubmed ID: #{self.locator_id}")
+      paper.errors.delete(:title)
```

Comment:
I should probably rewrite this to make it more clear.  The current mapping creates a new Paper and imports crossref and Pubmed data if it can locate it in their api's.  If it can't find it in either, it will return a paper with a blank title which won't save.  However, this will also return a "title can't be blank" error, which I'm getting rid of.

I can probably add something to the "else" to show the title is nil because it didn't find anything in pubmed or crossref.  That would make it more readable.